### PR TITLE
Feature: dynamic bookmark, view and play counts

### PR DESCRIPTION
**Pull request by bertyhell**
closes:
* https://district01.atlassian.net/browse/AVO2-551
* https://district01.atlassian.net/browse/AVO2-106
* https://district01.atlassian.net/browse/AVO2-47

search results:
image

item detail:
image

collection detail:
image

bundle detail:
image

### CODE REVIEW

File: src/shared/services/bookmarks-views-plays-service.ts [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+	): Promise<boolean> {
+		try {
+			if (!contentId) {
+				throw new CustomError(
+					`Failed to bookmark ${type} because the ${type} doesn't seem to be loaded yet`,
+					null,
+					{ contentId }
+				);
+			}
+			if (isBookmarked) {
```
Comment (benjaminnaesen) [inline src/shared/services/bookmarks-views-plays-service.ts]: `await BookmarksViewsPlaysService.action(isBookmarked ? 'unbookmark' : 'bookmark', type, contentId, user, false);` kan deze if-statement vervangen.

File: src/shared/services/bookmarks-views-plays-service.ts [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+			const mutation = get(EVENT_QUERIES, [action, contentTypeSimplified, 'query']);
+			const variables = get(EVENT_QUERIES, [action, contentTypeSimplified, 'variables'], () => {})(
+				contentUuid,
+				get(user, 'profile.id')
+			);
+			if (!mutation || !variables) {
+				throw new CustomError('Failed to find query/variables in query lookup table');
+			}
+			console.log('query', print(mutation));
```
Comment (benjaminnaesen) [inline src/shared/services/bookmarks-views-plays-service.ts]: Hans! Get ze flamethrower.

File: src/shared/services/bookmarks-views-plays-service.ts [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+			console.log('variables', variables);
```
Comment (benjaminnaesen) [inline src/shared/services/bookmarks-views-plays-service.ts]: Hanz, you missed a spot.

File: src/search/components/SearchResultItem.tsx [hunk 4]
```diff
@@ -28,8 +28,8 @@ const SearchResultItem: FunctionComponent<SearchResultItemProps> = ({
 			date={formatDate(result.dcterms_issued)}
 			// TODO: DISABLED_FEATURE
 			// tags={[{ label: 'Redactiekeuze', id: 'redactiekeuze' }, { label: 'Partner', id: 'partner' }]}
-			viewCount={412}
-			bookmarkCount={85}
+			viewCount={(result as any).views_count || 0}
```
Comment (benjaminnaesen) [inline src/search/components/SearchResultItem.tsx]: Mij lijkt het dat we hier wel een type kunnen voorzien. Voorlopig een TODO bijzetten?

File: src/item/views/ItemDetail.tsx [hunk 5]
```diff
@@ -372,8 +394,9 @@ const ItemDetail: FunctionComponent<ItemDetailProps> = ({
 											<ToggleButton
 												type="tertiary"
 												icon="bookmark"
-												active={false}
+												active={bookmarkViewPlayCounts.isBookmarked}
 												ariaLabel={t('item/views/item___toggle-bladwijzer')}
+												onClick={() => toggleBookmark()}
```
Comment (benjaminnaesen) [inline src/item/views/ItemDetail.tsx]: `onClick={toggleBookmark}`

File: src/item/components/modals/AddToCollectionModal.tsx [hunk 6]
```diff
@@ -46,6 +45,7 @@ import { getThumbnailForCollection } from '../../../shared/services/stills-servi
 import toastService from '../../../shared/services/toast-service';
 
 import './AddToCollectionModal.scss';
+import FlowPlayerWrapper from '../../../shared/components/FlowPlayerWrapper/FlowPlayerWrapper';
```
Comment (benjaminnaesen) [inline src/item/components/modals/AddToCollectionModal.tsx]: Import order.

Comment (bertyhell) [reply src/item/components/modals/AddToCollectionModal.tsx]: fixed in: feature/bookmarks-in-workspace

File: src/item/views/ItemDetail.tsx (hunk 5 above)
Comment (bertyhell) [reply src/item/views/ItemDetail.tsx]: fixed in: feature/bookmarks-in-workspace

File: src/search/components/SearchResultItem.tsx (hunk 4 above)
Comment (bertyhell) [reply src/search/components/SearchResultItem.tsx]: fixed in: feature/bookmarks-in-workspace

File: src/shared/services/bookmarks-views-plays-service.ts (hunk 3 above)
Comment (bertyhell) [reply src/shared/services/bookmarks-views-plays-service.ts]: fixed in: feature/bookmarks-in-workspace

File: src/shared/services/bookmarks-views-plays-service.ts (hunk 2 above)
Comment (bertyhell) [reply src/shared/services/bookmarks-views-plays-service.ts]: fixed in: feature/bookmarks-in-workspace

File: src/shared/services/bookmarks-views-plays-service.ts (hunk 1 above)
Comment (bertyhell) [reply src/shared/services/bookmarks-views-plays-service.ts]: fixed in: feature/bookmarks-in-workspace

**Pull request closed without merging**